src/java/main/org/apache/zookeeper/ClientCnxn.java

```diff
@@ -1102,7 +1103,32 @@ private void startConnect(InetSocketAddress addr) throws IOException {
         private String getServerPrincipal(InetSocketAddress addr) {
             String principalUserName = clientConfig.getProperty(ZKClientConfig.ZK_SASL_CLIENT_USERNAME,
                     ZKClientConfig.ZK_SASL_CLIENT_USERNAME_DEFAULT);
-            String serverPrincipal = principalUserName + "/" + addr.getHostString();
+            String hostName = addr.getHostName();
+
+            boolean canonicalize = true;
+            try {
+                canonicalize = Boolean.parseBoolean(System.getProperty(ZKClientConfig.ZK_SASL_CLIENT_CANONICALIZE_HOSTNAME,
```

Comment:
Isn't it clientConfig instead of System ?

Reply:
Great point, missed that on the port.